project/src/main/java/com/google/autograder/data/Database.java

```diff
@@ -1,151 +1,309 @@
+                courseEntity.setProperty("name", course.get("name"));
+                courseEntity.setProperty("description", course.get("description"));
+                courseEntity.setProperty("creationTime", course.get("creationTime"));
+                courseEntity.setProperty("userEmail", userEmail);
+                courseEntity.setProperty("userID", userID);
+
+                DATA_STORE.put(courseEntity);
+            }
+        }        
+    }
+
+    // Retrives the current user's courses data as JSON.
+
+    public static String getCoursesData() {
+        String userEmail = UserHandler.getCurrentUserEmail();
+
+        Filter userEmailFilter = new FilterPredicate("userEmail", FilterOperator.EQUAL, userEmail);
+        Query query = new Query("Course").setFilter(userEmailFilter).addSort("creationTime", SortDirection.DESCENDING);
+        PreparedQuery results = DATA_STORE.prepare(query);
+
+        StringBuilder coursesData = new StringBuilder("{\"courses\": [");
+
+        for (Entity course : results.asIterable()) {
+            String name = (String) course.getProperty("name");
+            String courseID = (String) course.getProperty("id");
+
+            StringBuilder courseData = new StringBuilder("{");
+            
+            courseData.append("\"name\":" + "\"" + name + "\"");
+            courseData.append(",");
+            courseData.append("\"id\":" + "\"" + courseID + "\"");
+            courseData.append("},");
+
+            coursesData.append(courseData.toString());
+        }
 
-  public final DatastoreService datastore;
-  public final List<String> answerGradingStatuses;
+        coursesData = new StringBuilder(coursesData.toString().substring(0, coursesData.length() - 1));
+        coursesData.append("]}");
+        
+        return coursesData.toString();
+    }
+
+    // Takes in a course work array in JSON format and then stores the assignments data for the current user.
+
+    public static void storeAssignmentsData(String courseWorkJSON, String courseID) {        
+        Filter courseIDFilter = new FilterPredicate("courseID", FilterOperator.EQUAL, courseID);
+        Query query = new Query("Assignment").setFilter(courseIDFilter);
+        PreparedQuery results = DATA_STORE.prepare(query);
+
+        // TODO: Compare Datastore contents with the course work data from Google Classroom
+
+        for (Entity assignment : results.asIterable()) {
+            DATA_STORE.delete(assignment.getKey());
+        }
+
+        Iterator courseWorkIterator = null;
+
+        try {
+            JSONObject courseWorkObject = (JSONObject) new JSONParser().parse(courseWorkJSON); 
+            JSONArray courseWork = (JSONArray) courseWorkObject.get("courseWork");
+            courseWorkIterator = courseWork.iterator();
+        } catch (ParseException exception) {
+            System.out.println("\n\n" + "Error parsing JSON Course Work data from the Google Classroom API" + "\n\n");
```

Comment:
Similarly, return here to get rid of the if statement below

Reply:
+1